src/io/mod.rs

```diff
@@ -48,3 +47,357 @@ mod stdin;
+            ///
+            /// The default implementation calls `write` with either the first nonempty
+            /// buffer provided, or an empty one if none exists.
+            fn write_vectored(&mut self, bufs: &[IoSlice<'_>]) -> Result<usize> {
+                unimplemented!()
+            }
+
+            /// Flush this output stream, ensuring that all intermediately buffered
+            /// contents reach their destination.
+            ///
+            /// # Errors
+            ///
+            /// It is considered an error if not all bytes could be written due to
+            /// I/O errors or EOF being reached.
+            ///
+            /// # Examples
+            ///
+            /// ```no_run
+            /// use std::io::prelude::*;
+            /// use std::io::BufWriter;
+            /// use std::fs::File;
+            ///
+            /// fn main() -> std::io::Result<()> {
+            ///     let mut buffer = BufWriter::new(File::create("foo.txt")?);
+            ///
+            ///     buffer.write_all(b"some bytes")?;
+            ///     buffer.flush()?;
+            ///     Ok(())
+            /// }
+            /// ```
+            fn flush(&mut self) -> Result<()>;
+
+            /// Attempts to write an entire buffer into this writer.
+            ///
+            /// This method will continuously call [`write`] until there is no more data
+            /// to be written or an error of non-[`ErrorKind::Interrupted`] kind is
+            /// returned. This method will not return until the entire buffer has been
+            /// successfully written or such an error occurs. The first error that is
+            /// not of [`ErrorKind::Interrupted`] kind generated from this method will be
+            /// returned.
+            ///
+            /// # Errors
+            ///
+            /// This function will return the first error of
+            /// non-[`ErrorKind::Interrupted`] kind that [`write`] returns.
+            ///
+            /// [`ErrorKind::Interrupted`]: ../io/enum.ErrorKind.html#variant.Interrupted
+            /// [`write`]: #tymethod.write
+            ///
+            /// # Examples
+            ///
+            /// ```no_run
+            /// use std::io::prelude::*;
+            /// use std::fs::File;
+            ///
+            /// fn main() -> std::io::Result<()> {
+            ///     let mut buffer = File::create("foo.txt")?;
+            ///
+            ///     buffer.write_all(b"some bytes")?;
+            ///     Ok(())
+            /// }
+            /// ```
+            fn write_all(&mut self, mut buf: &[u8]) -> Result<()> {
+                unimplemented!()
+            }
+
+            /// Writes a formatted string into this writer, returning any error
+            /// encountered.
+            ///
+            /// This method is primarily used to interface with the
+            /// [`format_args!`] macro, but it is rare that this should
```

Comment:
I couldn't quite figure out how to get this approach to work for macros but I might be missing something simple.

For reference, in the docs for `io::Write` the macros are linked like this:

```rust
    /// This method is primarily used to interface with the
    /// [`format_args!`][formatargs] macro, but it is rare that this should
    /// explicitly be called. The [`write!`][write] macro should be favored to
    /// invoke this method instead.
    ///
    /// [formatargs]: ../macro.format_args.html
    /// [write]: ../macro.write.html
```

However, when I try to define those macros locally (either within `src/io/mod.rs` or within `src/lib.rs`) the `cargo deadlinks` still complains.